# Initial Data Inspection

This notebook performs an initial inspection of the employee salary dataset. The purpose is to understand the dataset's structure, identify potential data quality issues, and gather summary information before data cleaning and machine learning model development.

## Step 1: Import Required Libraries

In [1]:
# Import the Pandas library for data loading and analysis
import pandas as pd

# Load the original salary dataset from the raw data folder
df = pd.read_csv("../data/raw/ds_salaries.csv")

# Display the first five rows to verify the dataset loaded correctly
df.head()

,Unnamed: 0,work_year,experience_level,employment_type,job_title,salary,salary_currency,salary_in_usd,employee_residence,remote_ratio,company_location,company_size
0,0,2020,MI,FT,Data Scientist,70000,EUR,79833,DE,0,DE,L
1,1,2020,SE,FT,Machine Learning Scientist,260000,USD,260000,JP,0,JP,S
2,2,2020,SE,FT,Big Data Engineer,85000,GBP,109024,GB,50,GB,M
3,3,2020,MI,FT,Product Data Analyst,20000,USD,20000,HN,0,HN,S
4,4,2020,SE,FT,Machine Learning Engineer,150000,USD,150000,US,50,US,L


## Step 2: Determine the Dataset Size

In [2]:
# Display the number of rows and columns in the dataset
print("Rows:", df.shape[0])
print("Columns:", df.shape[1])

Rows: 607
Columns: 12


## Step 3: Display the Dataset Columns

In [3]:
# Display the names of all columns in the dataset
print(df.columns.tolist())

['Unnamed: 0', 'work_year', 'experience_level', 'employment_type', 'job_title', 'salary', 'salary_currency', 'salary_in_usd', 'employee_residence', 'remote_ratio', 'company_location', 'company_size']


## Step 4: Inspect Data Types and Missing Values

In [4]:
# Display information about each column including data type and non-null values
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 607 entries, 0 to 606
Data columns (total 12 columns):
 #   Column              Non-Null Count  Dtype
---  ------              --------------  -----
 0   Unnamed: 0          607 non-null    int64
 1   work_year           607 non-null    int64
 2   experience_level    607 non-null    str  
 3   employment_type     607 non-null    str  
 4   job_title           607 non-null    str  
 5   salary              607 non-null    int64
 6   salary_currency     607 non-null    str  
 7   salary_in_usd       607 non-null    int64
 8   employee_residence  607 non-null    str  
 9   remote_ratio        607 non-null    int64
 10  company_location    607 non-null    str  
 11  company_size        607 non-null    str  
dtypes: int64(5), str(7)
memory usage: 73.8 KB


## Step 5: Check for Missing Values

In [5]:
# Count the number of missing values in each column
df.isnull().sum().sort_values(ascending=False)

Unnamed: 0            0
work_year             0
experience_level      0
employment_type       0
job_title             0
salary                0
salary_currency       0
salary_in_usd         0
employee_residence    0
remote_ratio          0
company_location      0
company_size          0
dtype: int64

## Step 6: Check for Duplicate Records

In [6]:
# Count duplicate rows in the dataset
duplicate_count = df.duplicated().sum()

print("Duplicate rows:", duplicate_count)

Duplicate rows: 0


## Step 7: Generate Summary Statistics

In [7]:
# Generate summary statistics for both numerical and categorical features
df.describe(include="all").T

,count,unique,top,freq,mean,std,min,25%,50%,75%,max
Unnamed: 0,607.0,NaN,NaN,NaN,303.0,175.370085,0.0,151.5,303.0,454.5,606.0
work_year,607.0,NaN,NaN,NaN,2021.405272,0.692133,2020.0,2021.0,2022.0,2022.0,2022.0
experience_level,607,4,SE,280,NaN,NaN,NaN,NaN,NaN,NaN,NaN
employment_type,607,4,FT,588,NaN,NaN,NaN,NaN,NaN,NaN,NaN
job_title,607,50,Data Scientist,143,NaN,NaN,NaN,NaN,NaN,NaN,NaN
salary,607.0,NaN,NaN,NaN,324000.062603,1544357.486636,4000.0,70000.0,115000.0,165000.0,30400000.0
salary_currency,607,17,USD,398,NaN,NaN,NaN,NaN,NaN,NaN,NaN
salary_in_usd,607.0,NaN,NaN,NaN,112297.869852,70957.259411,2859.0,62726.0,101570.0,150000.0,600000.0
employee_residence,607,57,US,332,NaN,NaN,NaN,NaN,NaN,NaN,NaN
remote_ratio,607.0,NaN,NaN,NaN,70.92257,40.70913,0.0,50.0,100.0,100.0,100.0


## Step 8: Examine the Target Variable

In [8]:
# Display descriptive statistics for the salary prediction target variable
df["salary_in_usd"].describe()

count       607.000000
mean     112297.869852
std       70957.259411
min        2859.000000
25%       62726.000000
50%      101570.000000
75%      150000.000000
max      600000.000000
Name: salary_in_usd, dtype: float64

## Step 9: Count Unique Values

In [9]:
# Display the number of unique values contained in each feature
for column in df.columns:
    print(f"{column}: {df[column].nunique()} unique values")

Unnamed: 0: 607 unique values
work_year: 3 unique values
experience_level: 4 unique values
employment_type: 4 unique values
job_title: 50 unique values
salary: 272 unique values
salary_currency: 17 unique values
salary_in_usd: 369 unique values
employee_residence: 57 unique values
remote_ratio: 3 unique values
company_location: 50 unique values
company_size: 3 unique values


# Initial Data Findings

### Summary

- The dataset contains **607 employee records** and **12 columns**.
- No missing values were identified.
- No duplicate records were found.
- The dataset contains a combination of numerical and categorical variables.
- The target variable for this project is **salary_in_usd**.
- Employee salaries range from **$2,859** to **$600,000**.
- The dataset contains **50 unique job titles**, **50 company locations**, and **57 employee residence locations**.
- Most predictor variables are categorical and will require encoding before machine learning models can be trained.